Dockerfile

```
@@ -10,17 +10,24 @@ COPY go.sum go.sum
 # and so that source changes don't invalidate our downloaded layer
 # RUN go mod download
 
+COPY Makefile Makefile
+COPY Makefile.helm.mk Makefile.helm.mk
+COPY Makefile.helper.mk Makefile.helper.mk
+COPY Makefile.specialresource.mk Makefile.specialresource.mk
+COPY hack/ hack/
+
 # Copy the go source
 COPY main.go main.go
 COPY api/ api/
 COPY cmd/ cmd/
 COPY controllers/ controllers/
+COPY .patches/ .patches/
 
 COPY pkg/ pkg/
 COPY vendor/ vendor/
 
 # Build
-RUN CGO_ENABLED=0 GO111MODULE=on go build -mod=vendor -a -o manager main.go
+RUN make manager
```

Shall we use exec-style here to avoid spawning a shell?
```suggestion
RUN ["make", "manager"]
```